datastore.go

```diff
@@ -117,17 +117,36 @@ func (a *accessor) Query(q dsq.Query) (dsq.Results, error) {
 	return a.queryNew(q)
 }
 
+// rangeForQuery returns a new Range struct that starts and bounds a query,
+// based on Query.Prefix and Query.SeekPrefix
+func rangeForQuery(q *dsq.Query) *util.Range {
+	var rnge *util.Range
+
+	if len(q.Prefix) > 0 {
+		rnge = util.BytesPrefix([]byte(q.Prefix))
+	}
+
+	if len(q.SeekPrefix) > 0 {
```

Comment:
See https://github.com/ipfs/go-ds-badger/pull/47#discussion_r254396841. We should probably fail here if we're applying any order *other* than "order by key".